blueprint-upload-component-preview-cy-587

bredek opened the pull request:
![CleanShot 2022-01-04 at 12 03 00](https://user-images.githubusercontent.com/14944987/148056353-24c86c67-882c-452d-bde7-d394b2a7713d.gif)

Commit:
chore: setup boilerplate

chore: setup boilerplate

Commit:
chore: initalValue should be optional

chore: initalValue should be optional

bredek committed:
very dirty and rough component showcase

bredek committed:
Merge remote-tracking branch 'origin/dev' into blueprint-upload-component-preview-cy-587

bredek committed:
feat: modified Divider to accept width, height, vertical alignment

bredek committed:
feat: modified FlatActionButton to accept width, height, disabled state

bredek committed:
feat: updated SetImage card and mocked util functions

diegomart2000 reviewed (changes requested):
@bredek we will add the dropzone into this component.

MoritzGruber commented on src/components/set-image-card/styles/SSetImageCard.tsx:
```diff
@@ -0,0 +1,85 @@
+  box-shadow: var(--box-shadow-strong-pop-elavation);
+`;
+
+export const SSetImageButtonsBar = styled(SFlex)`
```
@bredek  
the ideal approach we try to use with styled components it to create a file for each of them 
so SSetImageButtonsBar or SSetImageButtonsBar would be a sperate file 

in order to speed up the process you can generate this files with prop like so:
 `yarn plop s SetImageCard ButtonsBar`

bredek replied:
Moved all probable styled components into a separate files

MoritzGruber commented on src/components/set-image-card/SetImageCard.tsx:
```diff
@@ -0,0 +1,101 @@
+        </SSetImageHeader>
+        {/* FIXME! */}
+        {/* Probably we should change img wrapper to existing element when url processing is done on server */}
+        {/* {image && <Img resolution={48} src={image} />} */}
```
@bredek  i think you can use our image wrapper already and there is no need to put this comment in even with some mock data urls like this 
https://images.pexels.com/photos/775201/pexels-photo-775201.jpeg?cs=srgb&dl=pexels-kaique-rocha-775201.jpg&fm=jpg
too

bredek replied:
Fixed here - https://github.com/caisy-io/league/pull/169/commits/77927a448205dd76fc80590f7b0cdc7091734463#r779508815

MoritzGruber commented:
also in regards to the dropzone
i think that should not block this component in league - all we need to do is give this component some wrapper with unique id or classname and a data prop with what we need for the upload
since we have a global catch of all drop events in caisy for the global image upload we need to handle this image upload there as well - so that is something we need to handle in web-app

@diegomart2000 and myself just had a chat @bredek  so after my change request we should be fine

bredek committed:
Merge branch 'dev' into blueprint-upload-component-preview-cy-587

bredek committed:
fix: PR fixes

- moved common styles to separate files
- mocked new uploadImage to return temporary url of image
- minor optimisations

bredek commented on src/components/set-image-card/SetImageCard.stories.tsx:
```diff
@@ -12,7 +12,7 @@ const processImageMock = (file: File) => {
     return new Promise<string>((resolve) => {
       setTimeout(() => {
-        resolve(imageUrl);
+        resolve("https://images.pexels.com/photos/775201/pexels-photo-775201.jpeg?cs=srgb&dl=pexels-kaique-rocha-775201.jpg&fm=jpg");
```
mocked temp url for proper `Img` component working 
https://github.com/caisy-io/league/pull/169#discussion_r778985041

bredek commented:
> also in regards to the dropzone i think that should not block this component in league - all we need to do is give this component some wrapper with unique id or classname and a data prop with what we need for the upload since we have a global catch of all drop events in caisy for the global image upload we need to handle this image upload there as well - so that is something we need to handle in web-app
> 
> @diegomart2000 and myself just had a chat @bredek so after my change request we should be fine

So what's your final call? Should I add dropzone in this particular component or we can handle this outside in the web app?

MoritzGruber commented on src/components/set-image-card/SetImageCard.tsx:
```diff
@@ -0,0 +1,101 @@
+
+  const imageRef = useRef<HTMLInputElement>(null);
+
+  const mockProcessIamge = () => {
```
this function name `mockProcessIamge` is a bit off to me ^^ - i would rather name it `triggerImageSelection`
or something

bredek replied:
you're right! forgot this one, will `openImagePicker` work?

MoritzGruber commented:
So what's your final call? Should I add dropzone in this particular component or we can handle this outside in the web app?

-> do not add  dropzone - we handle it in web-app from outside

bredek committed:
chore: naming refactoring

diegomart2000 commented on src/components/flat-action-button/FlatActionButton.tsx:
```diff
@@ -6,11 +6,21 @@ export type IFlatActionButtonType = "white" | "blue" | "success" | "danger" | "g
+}) => {
   return (
-    <SFlatActionButton type={type} onClick={onClick}>
+    <SFlatActionButton disabled={disabled} type={type} width={width} height={height} onClick={onClick}>
```
@bredek this shouldn't be changed like this, otherwise we are breaking the definitions in DS. Why do we need this?

diegomart2000 replied:
Looks like the component should use a different component. https://www.figma.com/file/1hz5D4Q7pj5w0YrRw7hXbu/UI-Kit?node-id=887%3A13505

diegomart2000 replied:
this component should be using Sticked button, not this one, and also the changes to the flat action button should be reverted.

![Uploading Screen Shot 2022-01-06 at 10.55.05 AM.png…]()

bredek replied:
But we don't have any sticked button in our library.
We had a talk regarding this button with team.

![image](https://user-images.githubusercontent.com/14944987/148399779-d3ba9e34-ff83-472e-b268-2d828a98835f.png)

diegomart2000 replied:
Yes, it is in flight https://linear.app/caisy/issue/CY-632/buttons-bar

diegomart2000 replied:
https://league-9zpvcb3b8-caisy.vercel.app/?path=/story/components-buttons-button--default&args=content:DEFAULT;sticked:true

diegomart2000 replied:
https://linear.app/caisy/issue/CY-632/buttons-bar Already merged into dev

diegomart2000 commented on src/components/set-image-card/SetImageCard.tsx:
```diff
@@ -0,0 +1,101 @@
+      </SSetImageCardBody>
+      {image && (
+        <SSetImageCardButtonsBar>
+          <FlatActionButton disabled={isLoading} type="default" onClick={openImagePicker}>
```
This is the wrong component, see comments above.

diegomart2000 commented on src/components/set-image-card/SetImageCard.tsx:
```diff
@@ -0,0 +1,101 @@
+            CHANGE PREVIEW
+          </FlatActionButton>
+          <Divider marginBottom={0} marginTop={0} vertical width={44} />
+          <FlatActionButton disabled={isLoading} type="danger" onClick={removeImage}>
```
Same.

diegomart2000 commented on src/components/set-image-card/SetImageCard.tsx:
```diff
@@ -0,0 +1,101 @@
+        <input hidden type="file" ref={imageRef} onChange={uploadImage} />
+      </SSetImageCardBody>
+      {image && (
+        <SSetImageCardButtonsBar>
```
There is a buttons bar component from DS that should be used here.

bredek replied:
And we don't have buttons bar component as well.

diegomart2000 replied:
https://linear.app/caisy/issue/CY-632/buttons-bar Already merged into dev

diegomart2000 commented on src/components/set-image-card/index.ts:
```diff
@@ -0,0 +1 @@
+export { SetImageCard } from "./SetImageCard";
```
Please use DS image card component.

MoritzGruber replied:
@diegomart2000  this makes no sense to me since the the is not image card component used inside the set image card

bredek committed:
Merge branch 'dev' into blueprint-upload-component-preview-cy-587

bredek committed:
chore: revert flat button changes

bredek committed:
fix: added new buttons bar component

bredek committed:
feat: tag list item mock

bredek committed:
Merge branch 'dev' into blueprint-upload-component-preview-cy-587

bredek committed:
chore: PR fixes

bredek committed:
Merge branch 'dev' into blueprint-upload-component-preview-cy-587
